### ISSUE TITLE
Public Diplomacy films are broken and image formatting is off

### ISSUE THREAD

**Issue by joewiz** [HistoryAtState/hsg-shell#266]
Reported via @awmarrs, who will fill in details in subsequent comments.

**Comment by joewiz** [comment]
Background: Each item in Appendix A of https://history.state.gov/historicaldocuments/frus1917-72PubDip is a film or an image. 

Problems:

1. Video is missing from https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-5 and https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-11. We previously got the video working via https://github.com/HistoryAtState/hsg-shell/issues/18. 

2. Less concerning is the quirky formatting of the photos in the appendix, e.g., https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-1. Notice that (1) the footnote appears above the image instead of at the foot of the page and (2) "Contents" appears beneath the image. 

@HistoryAtState/existsolutions Could you please assist in restoring the films?

**Comment by awmarrs** [comment]
Hello,

The problems appear to be as follows:

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-5 - video missing and footnote text appears in the wrong place. Footnote link after "Appendix 5" is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-6 - shouldn't we have a full pdf of the document here? Footnote text in the wrong place and footnote link is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-7 - footnote text in the wrong place and footnote link is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-8 - should we have a full pdf of the document here? Footnote text in the wrong place and footnote link is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-10 - This is not the complete document. Footnote text in the wrong place and footnote link is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-11 - videos are missing. Footnote text in the wrong place and footnote link is not active.

https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-12 - did we have a full pdf for this document? Footnote text is in the wrong place and footnote link is not active.

**Comment by joewiz** [comment]
Thanks for the additional details @awmarrs. I agree:

Video is missing in appendixes 5 and 11
Footnote text appears in the wrong place and is inactive in all appendixes
Links to downloadable PDFs of pamphlets are missing in appendixes 6, 8, and 12

@tuurma The section of the file in question is:  https://github.com/HistoryAtState/frus/blob/master/volumes/frus1917-72PubDip.xml#L4780-L5361.

**Comment by joewiz** [comment]
@HistoryAtState/existsolutions Could you please update me on the status of this request?

**Comment by joewiz** [comment]
@HistoryAtState/existsolutions Testing - do you see this Magda?

**Comment by joewiz** [comment]
Update: @wolfgangmm and @tuurma [found the change needed](https://github.com/HistoryAtState/hsg-shell/commit/2fdf93d4056e6ddffd32921323d5c097c7b72217) to stop TEI PM from blocking the HTML embedded player code. This helps only the video issues above, and fixes them only on localhost, not hsg. Despite [some progress](https://github.com/HistoryAtState/frus/commit/e14abf8d1fa040e716b8c39bcc8cec0ad6896698), further video fixes await help from PA/WM, who can hopefully provide us with updated, working code from brightcove. 

The remaining non-video issues have not been addressed.

**Comment by joewiz** [comment]
The video issues are now basically fixed. 

The remaining non-video issues have still not been addressed.

**Comment by tuurma** [comment]
I have checked the mentioned cases, looks to me like

- video issues are fixed
- footnotes are displayed correctly, both on hover and at the bottom of the page
- links to footnotes are present and working

Links to pdf documents have a form

```
https://s3.amazonaws.com/static.history.state.gov.s3.amazonaws.com/frus/frus1917-72PubDip/Document%20A.6.pdf
```
coming directly from the TEI source, e.g.
https://github.com/HistoryAtState/frus/blob/master/volumes/frus1917-72PubDip.xml#L5195

and result in an error of the kind

```<Error>
<Code>NoSuchBucket</Code>
<Message>The specified bucket does not exist</Message>
<BucketName>static.history.state.gov.s3.amazonaws.com</BucketName>
<RequestId>AE2A983E47AE5188</RequestId>
<HostId>
hsd+TbfsVJaSN4GjF48GkWi8MmdHjgKBIr0ITx1TWIvtQoRpIdSxd1+NY4+0diL489orS0T53oA=
</HostId>
</Error>
```
Nevertheless getting rid of the first part of the link (s3.amazonaws.com/) and changing it to

`https://static.history.state.gov.s3.amazonaws.com/frus/frus1917-72PubDip/Document%20A.6.pdf`
`
results in a working link. Should I just change the sources accordingly?

I would also appreciate a comment here:
>https://history.state.gov/historicaldocuments/frus1917-72PubDip/appendix-10 - This is not the complete document. Footnote text in the wrong place and footnote link is not active.

As far as I can see, the whole appendix-10 is rendered and also corresponding document 39 is present and displayed in full.

**Comment by tuurma** [comment]
fwiw there are exactly 6 refs in frus with this kind of link prefix

collection("/db/apps/frus")//tei:ref[starts-with(@target, 'https://s3.amazonaws.com/')]

**Comment by tuurma** [comment]
Weirdly enough, links from other volumes following same pattern do work fine. @joewiz I'd appreciate your comment if this is intentional.

cf https://s3.amazonaws.com/static.history.state.gov/frus/frus1961-63v10-12mSupp/pdf/booklet.pdf
https://history.state.gov/historicaldocuments/frus1961-63v10-12mSupp/front-matter

and https://history.state.gov/historicaldocuments/frus1961-63v07-09mSupp/front-matter

**Comment by tuurma** [comment]
agreed with @joewiz to fix tei sources in question; removing high priority label

**Comment by joewiz** [comment]
@tuurma Thanks for narrowing in on the URL. These 6 URLs were in fact malformed. Instead of:

`https://s3.amazonaws.com/static.history.state.gov.s3.amazonaws.com/frus/frus1917-72PubDip/Document%20A.6.pdf`

They should've been:

`https://s3.amazonaws.com/static.history.state.gov/frus/frus1917-72PubDip/Document%20A.6.pdf`

The 2nd `.s3.amazonaws.com` bit was the source of the error. Removing it results in the valid URL.

(Some background here: Our old DNS entry to static.history.state.gov worked when we just used HTTP connections to history.state.gov, but since we instituted HSTS on history.state.gov and its subdomains, browsers will refuse to recognize the wildcard SSL certificate for S3 as valid for a history.state.gov subdomain. So we have switched URLs to all of our S3 assets to the equivalent form that raises no SSL errors: `https://s3.amazonaws.com/static.history.state.gov/...`)

**Comment by joewiz** [comment]
I've reviewed the items in @awmarrs's original list of problems and confirmed they have all been fixed.

So, in summary, 3 factors were responsible for the problems we saw here:

1. The FRUS TEI ODD prevented the snippets from being passed through. Fixing the ODD files allowed the snippets to be passed through.
2. The embedded video player HTML snippets lacked a required SSL directive, so HSTS-respecting browsers were refusing to load the video players. Adding these directives allowed the video players to load.
3. The URLs to the downloadable PDFs were wrong, probably leading back to when we tried to adjust URLs to work in an SSL context. Fixing the URLs allowed the PDFs to work.

Thanks to @wolfgangmm for identifying number 1 and @tuurma for identifying number 3.
